compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -364,11 +364,12 @@ public BallerinaVariableSymbol createVariableSymbol(BVarSymbol symbol, String na
     }
 
     private boolean isReadonlyIntersectionArrayType(BType type) {
-        if (type.tag == TypeTags.INTERSECTION && type.tsymbol.getOrigin() == SymbolOrigin.VIRTUAL) {
+        if (type.tsymbol != null && type.tag == TypeTags.INTERSECTION 
+                && type.tsymbol.getOrigin() == SymbolOrigin.VIRTUAL) {
             return true;
         }
 
-        if (type.tag == TypeTags.ARRAY) {
+        if (type.tsymbol != null && type.tag == TypeTags.ARRAY) {
```

Comment:
Since we call `isReadonlyIntersectionArrayType` recursively, shall we put the `null` check at the top of the function body?